nautobot/core/settings_funcs.py

```diff
@@ -82,3 +84,84 @@ def parse_redis_connection(redis_database):
         return f"{redis_scheme}://{redis_creds}{redis_host}?db={redis_database}"
     else:
         return f"{redis_scheme}://{redis_creds}{redis_host}:{redis_port}/{redis_database}"
+
+
+def setup_structlog_logging(
+    django_logging: dict,
+    django_apps: list,
+    django_middleware: list,
+    log_level="INFO",
+    root_level="INFO",
+    debug=False,
+    debug_db=False,
+    plain_format=False,
+) -> None:
+    """Set up structlog logging for Django."""
+    django_logging["version"] = 1
+    django_logging["disable_existing_loggers"] = True
+    if "test" in sys.argv:
+        django_logging["handlers"] = {
+            "null_handler": {
+                "level": "INFO",
+                "class": "logging.NullHandler",
+            },
+        }
+        for logger in django_logging["loggers"].values():
+            logger["handlers"] = ["null_handler"]
+            logger["level"] = "INFO"
+
+    django_apps.append("django_structlog")
+    django_middleware.append("django_structlog.middlewares.RequestMiddleware")
+
+    processors = (
+        structlog.stdlib.add_log_level,
+        structlog.stdlib.add_logger_name,
+        structlog.stdlib.PositionalArgumentsFormatter(),
+        structlog.contextvars.merge_contextvars,
+        structlog.processors.TimeStamper(fmt="iso"),
+        structlog.processors.StackInfoRenderer(),
+        structlog.processors.UnicodeDecoder(),
+        *([] if debug else [structlog.processors.format_exc_info]),
+    )
+
+    django_logging["formatters"] = {
+        "default_formatter": {
+            "()": structlog.stdlib.ProcessorFormatter,
+            "foreign_pre_chain": processors,
+            "processor": structlog.dev.ConsoleRenderer() if plain_format else structlog.processors.JSONRenderer(),
+        },
+    }
+
+    django_logging["handlers"] = {
+        "default_handler": {
+            "level": "DEBUG",
+            "class": "logging.StreamHandler",
+            "formatter": "default_formatter",
+        },
+    }
+
+    django_logging["root"] = {
+        "handlers": ["default_handler"],
+        "level": root_level,
+    }
+
+    if debug_db:
+        django_logging["loggers"]["django.db.backends"] = {"level": "DEBUG"}
```

Comment:
Why is this specific logger handled specially?

Reply:
This is driven by `NAUTOBOT_LOG_DEBUG_DB` environment variable, inspired by management services configuration.

Added info about new environment variables to `changes/5687.added`.

Reply:
And the answer to `why` is to be able to selectively enable database queries logging.